hyperdrive/packages/signer/sign/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+use anyhow::{anyhow, Result};
+use hyperware_process_lib::net::{NetAction, NetResponse};
+use hyperware_process_lib::{await_message, call_init, Address, Message, Request, Response};
+use serde::{Deserialize, Serialize};
+
+wit_bindgen::generate!({
+    path: "target/wit",
+    world: "sign-sys-v0",
+    generate_unused_types: true,
+    additional_derives: [PartialEq, serde::Deserialize, serde::Serialize, process_macros::SerdeJsonInto],
+});
+
+#[derive(Debug, Serialize, Deserialize)]
+enum SignerRequest {
+    Sign(SignRequest),
+    Verify { from: Address, data: SignResponse },
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct SignRequest {
+    pub site: String,
+    pub time: u64,
+    pub nonce: Option<String>,
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct SignResponse {
+    pub body: SignRequest,
+    pub message: Vec<u8>,
+    pub signature: Vec<u8>,
+}
+
+call_init!(initialize);
+fn initialize(our: Address) {
+    loop {
+        let msg = await_message();
+        match msg {
+            Err(_send_error) => {
+                // ignore send errors, local-only process
+                continue;
+            }
+            Ok(Message::Request { body, .. }) => handle_request(&our, &body).unwrap_or_default(),
+            _ => continue, // ignore responses
+        }
+    }
+}
```

Comment:
Preference for an `initialize` function more of this form: https://github.com/hyperware-ai/kit/blob/master/src/new/templates/rust/no-ui/chat/chat/src/lib.rs#L84-L92

We should be logging in error cases (which should never happen, but if they do, we want a record of them)